dcos/http.py

```diff
@@ -79,8 +86,51 @@ def request(method,
                 response.status_code,
                 response.text)
 
+    return response
+
+
+def request(method,
+            url,
+            is_success=_default_is_success,
+            timeout=None,
+            **kwargs):
+    """Sends an HTTP request. If the server responds with a 401, ask the
+    user for their credentials, and try request again (up to 3 times).
+
+    :param method: method for the new Request object
+    :type method: str
+    :param url: URL for the new Request object
+    :type url: str
+    :param is_success: Defines successful status codes for the request
+    :type is_success: Function from int to bool
+    :param timeout: request timeout
+    :type timeout: int
+    :param kwargs: Additional arguments to requests.request
+        (see http://docs.python-requests.org/en/latest/api/#requests.request)
+    :type kwargs: dict
+    :rtype: Response
+    """
+
+    if 'headers' not in kwargs:
+        kwargs['headers'] = {'Accept': 'application/json'}
+
+    timeout = timeout or _timeout()
+
+    response = _request(method, url, is_success, timeout, **kwargs)
+
+    i = 0
+    while i < 3 and response.status_code == 401:
+        _set_http_auth_credentials(response)
```

Comment:
fixed. If we have more than 1 commands, we cache in the global dict, so we don't ask for creds several times. Verified with some marathon adds/updates. I'll add some tests for this. 

We don't need BasicAuth to merge this PR, it will just be dead code. As was going to wait for it though, to make sure me and seb's implementations work together.